### ISSUE TITLE
Adding Points Layer with out_of_slice_display and shown

### ISSUE THREAD

**Issue by jduffield65** [napari/napari#4816]
## 🐛 Bug

I can't seem to add a points layer which uses both the `shown` and `out_of_slice_display` parameters.

## To Reproduce

If I run the following code:
```
import napari
import numpy as np

viewer = napari.Viewer()
points_zyx = np.array([[0, 1, 5], [1, 8, 9]])
shown = [False, True]
viewer.add_points(points_zyx, size=[3, 2, 2], shown=shown, out_of_slice_display=True)
napari.run()
```
I get an error:
<img width="1124" alt="napari_bug" src="https://user-images.githubusercontent.com/44460382/179214154-6a7520aa-7e77-4818-be08-7c3e69633a73.png">

But it does work if I set `shown=[True, True]`.

## Expected behavior

To be able to add the points layer but only showing some of the points, while at the same time having `out_of_slice_display=True`.

## Environment
```
napari: 0.4.16
Platform: macOS-12.4-arm64-arm-64bit
System: MacOS 12.4
Python: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33) [Clang 13.0.1 ]
Qt: 5.15.3
PyQt5: 5.15.4
NumPy: 1.23.0
SciPy: 1.8.1
Dask: 2022.6.1
VisPy: 0.10.0

OpenGL:
- GL version: 2.1 Metal - 76.3
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1728x1117, scale 2.0
- screen 2: resolution 1920x1080, scale 1.0
- screen 3: resolution 1920x1080, scale 1.0

Plugins:
- console: 0.0.4
- napari-svg: 0.1.6
- scikit-image: 0.4.16
```

**Comment by jni** [comment]
Thanks for the bug report @jduffield65!

**Comment by mstabrin** [comment]
Hello @jduffield65, could you install the current version of the `main` branch and test it there?
I think I fixed this issue some weeks ago :)

PR: https://github.com/napari/napari/pull/4699

**Comment by brisvag** [comment]
Closed by #4699.

**Comment by kcpevey** [comment]
Confirmed that the above code works for me on current main. Can we close @brisvag ?
